storage/remote/write.go

```diff
@@ -81,6 +88,8 @@ func NewWriteStorage(logger log.Logger, reg prometheus.Registerer, walDir string
 
 func (rws *WriteStorage) run() {
 	ticker := time.NewTicker(shardUpdateDuration)
+	rws.startTimedRecording()
```

Comment:
this would need to be run in a go routine, and the function should take a context or other way it can be shutdown/cleaned nicely